trl/trainer/grpo_config.py

```diff
@@ -412,3 +419,20 @@ class GRPOConfig(TrainingArguments):
             "all prompts are logged."
         },
     )
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        # The current default effective batch size
+        if self.num_mini_batches is None:
+            self.num_mini_batches = self.gradient_accumulation_steps
+
+        if self.num_mini_batches < self.gradient_accumulation_steps:
+            raise ValueError(
+                f"steps_per_generation ({self.num_mini_batches}) must be greater than or equal to gradient_accumulation_steps ({self.gradient_accumulation_steps})."
+            )
+
+        if self.num_mini_batches % self.gradient_accumulation_steps != 0:
+            raise ValueError(
+                f"steps_per_generation ({self.num_mini_batches}) must be divisible by gradient_accumulation_steps ({self.gradient_accumulation_steps})."
```

Comment:
Is this really necessary? One could imagine that `self.gradient_accumulation_steps=2` and `self.num_mini_batches=3`, in which case generation would occur in the middle of a gradient accumulation cycle. This would be a bit odd because the off-policyness could vary within an effective batch, but I don't see why this would be a problem.

Reply:
Yes you are right, we can relax this.